owned-manifests/machine-api-controllers.yaml

```diff
@@ -84,3 +84,15 @@ spec:
             cpu: 10m
             memory: 20Mi
       {{end}}
+      {{if .Controllers.ProviderRequestEnabled }}
+      - name: baremetal-opertor
```

Comment:
Do we expect to have provider requests for things other than the baremetal-operator? Should the data returned from getAdditionalProviderImages more completely describe the contents that need to be inserted into this template? Perhaps having them be full YAML snippets instead of just the image name would let us do that flexibly?